docs/src/content/docs/patterns/structured-outputs.mdx

```diff
@@ -47,51 +55,53 @@ const ExtractEntitiesBasic = gsx.Component<
     "organizations": ["org1", "org2", "org3"]
   }`;
   return (
-    <ChatCompletion
+    <GSXChatCompletion
       model="gpt-4o-mini"
       messages={[
         {
           role: "user",
           content: prompt,
         },
       ]}
-      response_format={{ type: "json_object" }}
-    >
-      {(response: string) => {
-        return JSON.parse(response) as ExtractEntitiesOutput;
-      }}
-    </ChatCompletion>
+      outputSchema={ExtractEntitiesSchema}
+    ></GSXChatCompletion>
   );
 });
 ```
 
-Using `response_format={{ type: "json_object" }}` ensure that we get valid JSON back, but it doesn't guarantee that the JSON will match the `ExtractEntitiesOutput` type. Parsing this way also doesn't provide any type safety.
-
-## Using Zod
+The `GSXChatCompletion` component will return the structured output directly matching the type of the `ExtractEntitiesSchema` with no extra parsing required.
 
-You can can take this one step further by using Zod to supply a JSON schema which guarantees that the model response will match the `ExtractEntitiesOutput` type.
+In the case that the model doesn't return valid JSON matching the schema, the task will be retried up to 3 times by default. You can change this behavior by setting the `retry` prop.
 
 ```tsx
-import { z } from "zod";
-import { zodResponseFormat } from "openai/helpers/zod";
+<GSXChatCompletion
+  model="gpt-4o-mini"
+  messages={[
+    {
+      role: "user",
+      content: prompt,
+    },
+  ]}
+  outputSchema={ExtractEntitiesSchema}
+  retry={{
```

Comment:
What is the default value here? I didn't think through this much when originally implementing, we should set something intentional.

Reply:
maxAttempts defaults to 3 meaning it will retry twice which seems reasonable to me.